Codec/Archive/Tar/Pack.hs

```diff
@@ -105,34 +110,58 @@ interleave = unsafeInterleaveIO . go
 -- * The file contents is read lazily.
 --
 packFileEntry :: FilePath -- ^ Full path to find the file on the local disk
-              -> TarPath  -- ^ Path to use for the tar Entry in the archive
-              -> IO Entry
+              -> These SplitError TarPath  -- ^ Path to use for the tar Entry in the archive
+              -> IO [Entry]
 packFileEntry filepath tarpath = do
-  mtime   <- getModTime filepath
-  perms   <- getPermissions filepath
-  file    <- openBinaryFile filepath ReadMode
-  size    <- hFileSize file
-  content <- BS.hGetContents file
-  return (simpleEntry tarpath (NormalFile content (fromIntegral size))) {
-    entryPermissions = if executable perms then executableFilePermissions
-                                           else ordinaryFilePermissions,
-    entryTime = mtime
-  }
+#if MIN_VERSION_directory(1,3,1)
+  entry <- pathIsSymbolicLink filepath
+    >>= \case
+    True -> do
+      linkTarget <- getSymbolicLinkTarget filepath
```

Comment:
I guess we have to do security check here too? Or dereference if the symlink points out of the base dir?